packages/next/build/webpack-config.ts

```diff
@@ -1272,6 +1273,10 @@ export default async function getBaseWebpackConfig(
       ].filter(Boolean),
     },
     plugins: [
+      ...getMiddlewareSourceMapPlugins({
+        isConfigured: !!config.experimental.middlewareSourceMaps,
+        isProductionBrowserSourceMapsOn: config.productionBrowserSourceMaps,
+      }),
```

Comment:
I would move the enable/disable logic directly into the config instead of passing aguments

```suggestion
      ...(config.experimental.middlewareSourceMaps && !config.productionBrowserSourceMaps
       ? getMiddlewareSourceMapPlugins()
       : []
      ),
```

Reply:
Maybe also add `!isServer` to only apply to the client compiler